docs/guides/vqa_guide/estimators.rst

```diff
@@ -4,46 +4,162 @@
+
+Let's look at the software that we built to accomodate all these!
+
+Estimation in Orquestra
+=======================
+
+As you can see there are many moving parts in the whole estimation process. Here we'll go through how they are all implemented in software.
+
+Data structures
+---------------
+
+There are two main data structures we use here. The first is ``EstimationTask``. It's an object which encapsulates all the data needed to perform estimation:
+
+- ``operator``: a   :class:`PauliSum <orquestra.quantum.operators.PauliSum>` which represents the observable
+- ``circuit``: a :class:`Circuit <orquestra.quantum.circuits.Circuit>` object which allow us to get a quantum state
+- ``number_of_shots``: ``int`` which specifies how many shots we make. 
+
+Another one is :class:`ExpectationValues <orquestra.quantum.measurements.ExpectationValues>`. Apart from storing the values themselves, if particular estimation method calculates correlations or covariances, it can also store them.
+
+Estimation method
+-----------------
+
+Once we have our :class:`EstimationTask <orquestra.quantum.api.estimation.EstimationTask>` defined, we can think about what algorithm for estimation to use. These algorithms are expressed using :class:`EstimateExpectationValues <orquestra.quantum.api.estimation.EstimateExpectationValues>` interface. It is implemented as Python Protocol (you can read more about it in :ref:`interfaces guide <interfaces_guide>`) and it's fairly simple:
+
+.. literalinclude:: ../../examples/estimators_guide.py
+    :start-after: >> Tutorial code snippet: estimation protocol
+    :lines: 2-6
+    :language: python
+
+
+Let's say you have created a new estimator called ``my_estimator``. In order to use it, you need to call it and provide a backend and a list of estimation_tasks. However, the definition you see above is not enough to make sure all the estimators work reliably. All the implementations of this protocol should also meet the following criteria (which we enforce by a set of contract tests):
+
+1. Return one ExpectationValue for each EstimationTask.
+2. The order in which ExpectationValues are returned should match the order in which EstimationTasks were provided.
+3. Number of entries in each ExpectationValue is not restricted - if the operator consists of multiple terms, estimator might be contain values for each term.
+4. Output ExpectationValue should include coefficients of the terms/operators.
+5. ``estimation_tasks`` can include tasks where operator consists of a constant term or contains a constant term. The implemented method should include the contributions of such constant terms in the return value.
+
+I hope points 1-3 are not controversial for you. Points 4-5 define a conventions that we have decided to follow - we know that in some circles people like using other conventions, which might be confusing. So we want to be clear about the conventions that we follow.
```

Comment:
Using the term "I" allowed in the documentation? If no, we can reword it as "An assumption is made here that points 1-3 are not controversial for the reader."

Reply:
I just changed it to "we", as I'm definitely using "we" across the docs.